File: 1647-minimum-deletions-to-make-character-frequencies-unique/1647-minimum-deletions-to-make-character-frequencies-unique.py
class Solution:
    def minDeletions(self, s: str) -> int:
        

        cntr = Counter(s) 

        seen = set() 
        cnt = 0 
        for k ,v in cntr.items(): 

            if v in seen:  
                while v!=0 and v in cntr.values(): 
                    v -=1 
                    cnt +=1 
                cntr[k] = v 
            else: 
                seen.add(v) 

        return cnt 

File: 1647-minimum-deletions-to-make-character-frequencies-unique/1647-minimum-deletions-to-make-character-frequencies-unique.cpp
class Solution {
public:
    int minDeletions(string s) {
        
    vector<int> fq (26 ,0) ; 
    for(auto e:s){ 
        fq[e-'a'] ++  ; } 
    sort(fq.begin() , fq.end()) ; 
    int ans = 0 ;
    for(int i=24 ; i >=0 ; i--){ 
        if( fq[i] ==0) break ; 
        
        if(fq[i] >=fq[i+1]) {
                int prv = fq[i] ; 
                fq[i] = max(0, fq[i+1] -1) ; 
                ans += prv - fq[i] ; 
                }
            }
        return ans ; 


        
        
    }
};